course/page/code.py

```diff
@@ -97,11 +98,19 @@ def debug_print(s):
                 "unix://var/run/docker.sock")
         docker_tls = getattr(settings, "RELATE_DOCKER_TLS_CONFIG",
                 None)
-        docker_cnx = docker.Client(
-                base_url=docker_url,
-                tls=docker_tls,
+
+        if platform.system().lower().startswith("linux"):
+            docker_cnx = docker.Client(
+                    base_url=docker_url,
+                    tls=docker_tls,
+                    timeout=docker_timeout,
+                    version="1.19")
+        else:
+            from docker.utils import kwargs_from_env
+            docker_cnx = docker.Client(
                 timeout=docker_timeout,
-                version="1.19")
+                **kwargs_from_env(assert_hostname=False)
```

Comment:
First I think I didn't make it clear that this PR is for developing on Windows or Mac OS using a local VM. The documentation of  ``kwargs_from_env()`` can be found [here](https://docker-py.readthedocs.io/en/1.2.3/boot2docker/), the kwarg ``assert_hostname=False`` is used because we are visiting docker from out side the VM, if that kwarg is missing, we will get an error ``CertificateError: hostname '192.168.99.10x' doesn't match 'localhost'``.

I also missed a instruction when using this, for windows, before we run ``python manage.py runserver``, we should run a shell cmd ``@FOR /f "tokens=*" %i IN ('docker-machine env default') DO @%i``. For Mac os, the cmd is ``$eval('docker-machine env default')``, so that ``kwargs_from_env()`` can get correct env variables for running docker.

The env variables on my machine are:

```
SET DOCKER_TLS_VERIFY=1
DOCKER_HOST=tcp://192.168.99.100:2376
DOCKER_CERT_PATH=d:\Users\dzhuang\.docker\machine\machines\default
DOCKER_MACHINE_NAME=default
```

Those variables can be get using cmd ``docker-machine env default`` when ``default`` is the name of the docker-machine.